### ISSUE TITLE
HTML errors in FundamentalWave

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1107]
**Reported by dietmarw on 16 May 2013 08:15 UTC**
The current version of HTML documentation in FundamentalWave contains the following errors (taken from [tidy.filtered](https://build.openmodelica.org/Documentation/tidy.filtered) file that is generated by OpenModelica on a daily basis)
```
[Checking file Modelica.Magnetic.FundamentalWave.UsersGuide.MultiPhase.html]:
line 26 column 7 - Warning: replacing unexpected b by </b>
line 26 column 62 - Warning: inserting implicit <font>
line 27 column 4 - Warning: inserting implicit <font>
line 28 column 4 - Warning: inserting implicit <font>
line 29 column 4 - Warning: inserting implicit <font>
line 30 column 4 - Warning: inserting implicit <font>
line 31 column 4 - Warning: inserting implicit <font>
line 32 column 4 - Warning: inserting implicit <font>
line 33 column 4 - Warning: inserting implicit <font>
line 34 column 4 - Warning: inserting implicit <font>
line 39 column 4 - Warning: inserting implicit <font>
line 44 column 4 - Warning: inserting implicit <font>
line 44 column 4 - Warning: missing </font> before <h4>
line 44 column 13 - Warning: inserting implicit <font>
line 44 column 22 - Warning: inserting implicit <p>
line 45 column 4 - Warning: inserting implicit <font>
line 47 column 2 - Warning: inserting implicit <font>
line 26 column 62 - Warning: trimming empty <font>
line 44 column 22 - Warning: trimming empty <p>
[Checking file Modelica.Magnetic.FundamentalWave.UsersGuide.Parameters.html]:
line 44 column 87 - Warning: inserting implicit <p>
line 44 column 87 - Warning: trimming empty <p>
\--
[Checking file Modelica.Magnetic.FundamentalWave.UsersGuide.html]:
line 25 column 1 - Warning: inserting implicit <p>
line 19 column 1 - Warning: trimming empty <p>
line 25 column 1 - Warning: trimming empty <p>
```

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1107

**Comment by modelica-trac-importer** [comment]
**Comment by christiankral on 18 May 2013 19:45 UTC**
Replying to [ticket:1107 dietmarw]:
> The current version of HTML documentation in FundamentalWave contains the following errors (taken from [tidy.filtered](https://build.openmodelica.org/Documentation/tidy.filtered) file that is generated by OpenModelica on a daily basis)
> ```
> [Checking file Modelica.Magnetic.FundamentalWave.UsersGuide.MultiPhase.html]:
> line 26 column 7 - Warning: replacing unexpected b by </b>

Personally, I cannot access Modelica.Magnetic.FundamentalWave.UsersGuide.MultiPhase.html (where is it located on the OpenModelica SVN repository or similar?). So I assume that line 26 of such HTML file means the following: If I store the HTML documentation of the affected Modelica class in a separate text file, then line 1 starts with `<html>`. Counting from there to line 26 does neither show me an **unexpected b** nor can I see a missing `</b>` which shall be inserted. 

> line 26 column 62 - Warning: inserting implicit <font>

How shall I interpret this message? `<font>` is not recommended in Modelica.UsersGuide.Conventions.Documentation at all.

> line 27 column 4 - Warning: inserting implicit <font>
> line 28 column 4 - Warning: inserting implicit <font>
> line 29 column 4 - Warning: inserting implicit <font>
> line 30 column 4 - Warning: inserting implicit <font>
> line 31 column 4 - Warning: inserting implicit <font>
> line 32 column 4 - Warning: inserting implicit <font>
> line 33 column 4 - Warning: inserting implicit <font>
> line 34 column 4 - Warning: inserting implicit <font>
> line 39 column 4 - Warning: inserting implicit <font>
> line 44 column 4 - Warning: inserting implicit <font>
> line 44 column 4 - Warning: missing </font> before <h4>
> line 44 column 13 - Warning: inserting implicit <font>
> line 44 column 22 - Warning: inserting implicit <p>
> line 45 column 4 - Warning: inserting implicit <font>
> line 47 column 2 - Warning: inserting implicit <font>
> line 26 column 62 - Warning: trimming empty <font>
> line 44 column 22 - Warning: trimming empty <p>
> [Checking file Modelica.Magnetic.FundamentalWave.UsersGuide.Parameters.html]:
> line 44 column 87 - Warning: inserting implicit <p>
> line 44 column 87 - Warning: trimming empty <p>

What is the rule on line 44 that we are not following?

> --
> [Checking file Modelica.Magnetic.FundamentalWave.UsersGuide.html]:
> line 25 column 1 - Warning: inserting implicit <p>
> line 19 column 1 - Warning: trimming empty <p>
> line 25 column 1 - Warning: trimming empty <p>
> ```

However, the recommended formatting of enumerated figures -- see Modelica.UsersGuide.Conventions.Documentation.Format.Figures -- has been incorporated in Modelica.Magnetic.FundamentalWave.UsersGuide.MultiPhase

**Comment by modelica-trac-importer** [comment]
**Comment by christiankral on 18 May 2013 19:53 UTC**
See da8db5e4571b217a17b8fd9568bd501d57d10dd6.

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 18 May 2013 21:49 UTC**
Replying to [comment:1 christiankral]:
> Personally, I cannot access Modelica.Magnetic.FundamentalWave.UsersGuide.MultiPhase.html (where is it located on the OpenModelica SVN repository or similar?). So I assume that line 26 of such HTML file means the following: If I store the HTML documentation of the affected Modelica class in a separate text file, then line 1 starts with `<html>`. Counting from there to line 26 does neither show me an **unexpected b** nor can I see a missing `</b>` which shall be inserted. 

The reported line numbers refer to the HTML that is extracted by OpenModelica's HTML generator. Basically this is just an indicator for what you need to look out. You can simply access the generated HTML documentation by browsing along starting from https://build.openmodelica.org/Documentation/. However you need to be aware that those HTML files will have been fixed by Tidy already but at least you can see what the HTML code was changed into.

You can and should not use the exported HTML of Dymola as reference since that one **adds** loads of HTML errors in addition unfortunately. 

What I normally do is simply counting the lines with the modelica info field of the respective class. This normally is almost in the right region. Then do a simple search for what ever was the offending tag. This normally works out quite nicely. To check if you found all errors you will either have to wait until next OpenModelica has rebuild the documentation (I think about once a day) or you can run my modified version locally on a Linux machine https://svn.modelica.org/projects/Modelica/branches/tools/MSLrelease/

> What is the rule on line 44 that we are not following?

All errors reported are real HTML issues and have nothing to do with any rule or guideline that we have defined in the User's Guide. So the errors reported are simply errors in the HTML syntax that need fixing.

I just ran the check locally on my machine and it seems you managed to fix it anyway :-)

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 18 May 2013 22:59 UTC**
Replying to [comment:3 dietmarw]:
> I just ran the check locally on my machine and it seems you managed to fix it anyway :-)

This wasn't entirely true. There was one tiny bit left that needed fixing. 
FIXED in ffcdfc9e1bbd700bd29ee32008c03dc87dac6c9a.
